.travis.yml

```diff
@@ -1,30 +1,21 @@
-matrix:
-  include:
-    - os: linux
-      dist: trusty
-      sudo: required
-      language: java
-    - os: osx
-      osx_image: xcode8.3
-      language: java
+dist: trusty
+sudo: required
+osx_image: xcode8.3
+language: java
+
+os:
+  - linux
+  - osx
 
 env:
-  global:
-    - BAZEL_VERSION=0.9.0
+  - BAZEL_VERSION=0.9.0
+  - BAZEL_VERSION=0.11.1
```

Comment:
how does this work? won't this overwrite the variable declared in `:11`?

Reply:
So when you specify a variable name multiple times, it converts those values into it to a [Build Matrix](https://docs.travis-ci.com/user/customizing-the-build#Build-Matrix). These then compound into the `os:` values to create a 2x2 set of test cases: [travis-ci.org/vsco/bazel-toolchains/builds/377357442](https://travis-ci.org/vsco/bazel-toolchains/builds/377357442?utm_source=github_status&utm_medium=notification)

Reply:
oh nice! I wasn't aware you could implicitly create build matrixes like this!